application/src/main/java/org/togetherjava/tjbot/features/help/HelpThreadCreatedListener.java

```diff
@@ -182,4 +186,17 @@ public void onModalSubmitted(ModalInteractionEvent event, List<String> args) {
         throw new UnsupportedOperationException("Not used");
     }
 
+    private void deletePinnedAnnouncement(ThreadChannel channel) {
+        channel.getHistoryFromBeginning(FIRST_TEN)
+            .map(historyFromBeginning -> historyFromBeginning.getRetrievedHistory()
+                .stream()
+                .filter(message -> message.getType() == MessageType.CHANNEL_PINNED_ADD)
+                .findFirst())
+            .flatMap(message -> message
+                .orElseThrow(() -> new IllegalStateException(
+                        "Unable to retrieve pinned message announcement"))
```

Comment:
i wouldnt give this error that much attention. its possible that it fails and not necessarily a bug. for example if the thread was insta-auto-deleted bc its scam.

in such a case, this "cleanup" code shouldnt generate unecessary noise that no one is interested in. id rather log it as `log.debug(...)`